### ISSUE TITLE
migration file "2.0.0-beta.1.js" failed

### ISSUE THREAD

**Issue by mitismirza** [Requarks/wiki#1097]
Os: Centos7
wiki.js version : 2.0.0-beta.303
Database Engine : mysql 
```
[root@localhost html]# mysql -V
mysql  Ver 15.1 Distrib 5.5.64-MariaDB, for Linux (x86_64) using readline 5.1
```
node version : v12.11.1
npm version : 6.11.3

hey guys , 

i'm trying to install wiki.js base on linux documention :  
https://docs.requarks.io/en/install/linux

until step 5 every thing is ok , but when i run ``` node server ``` it says : 
```
[root@localhost html]# node server
Loading configuration from /var/www/html/config.yml... OK
2019-10-14T15:12:25.439Z [MASTER] info: =======================================
2019-10-14T15:12:25.443Z [MASTER] info: = Wiki.js 2.0.0-beta.303 ==============
2019-10-14T15:12:25.443Z [MASTER] info: =======================================
2019-10-14T15:12:25.443Z [MASTER] info: Initializing...
2019-10-14T15:12:26.065Z [MASTER] info: Connecting to database...
2019-10-14T15:12:26.183Z [MASTER] info: Database Connection Successful [ OK ]
migration file "2.0.0-beta.1.js" failed
migration failed with error: create table `assets` (`id` int unsigned not null auto_increment primary key, `filename` varchar(255) not null, `basename` varchar(255) not null, `ext` varchar(255) not null, `kind` enum('binary', 'image') not null default 'binary', `mime` varchar(255) not null default 'application/octet-stream', `fileSize` int unsigned comment 'In kilobytes', `metadata` json, `createdAt` varchar(255) not null, `updatedAt` varchar(255) not null) default character set utf8mb4 - You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'json, `createdAt` varchar(255) not null, `updatedAt` varchar(255) not null) defa' at line 1
2019-10-14T15:12:26.322Z [MASTER] error: Database Initialization Error: create table `assets` (`id` int unsigned not null auto_increment primary key, `filename` varchar(255) not null, `basename` varchar(255) not null, `ext` varchar(255) not null, `kind` enum('binary', 'image') not null default 'binary', `mime` varchar(255) not null default 'application/octet-stream', `fileSize` int unsigned comment 'In kilobytes', `metadata` json, `createdAt` varchar(255) not null, `updatedAt` varchar(255) not null) default character set utf8mb4 - You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'json, `createdAt` varchar(255) not null, `updatedAt` varchar(255) not null) defa' at line 1
```

please help,

what i have to do?

**Comment by NGPixel** [comment]
Your mysql version is too old and not supported.

**Comment by centell** [comment]
I have same issue. my MySQL version is `5.6.40` using AWS RDS. what version of MySQL do i need?

node v12.13.0, npm 6.12.0

```
ubuntu@ip-10-0-1-115:~/wiki$ node server
Loading configuration from /home/ubuntu/wiki/config.yml... OK
2019-10-23T06:44:30.559Z [MASTER] info: =======================================
2019-10-23T06:44:30.561Z [MASTER] info: = Wiki.js 2.0.0-rc.1 ==================
2019-10-23T06:44:30.561Z [MASTER] info: =======================================
2019-10-23T06:44:30.561Z [MASTER] info: Initializing...
2019-10-23T06:44:30.983Z [MASTER] info: Connecting to database...
2019-10-23T06:44:31.432Z [MASTER] info: Database Connection Successful [ OK ]
migration file "2.0.0-beta.1.js" failed
migration failed with error: create table `assets` (`id` int unsigned not null auto_increment primary key, `filename` varchar(255) not null, `basename` varchar(255) not null, `ext` varchar(255) not null, `kind` enum('binary', 'image') not null default 'binary', `mime` varchar(255) not null default 'application/octet-stream', `fileSize` int unsigned comment 'In kilobytes', `metadata` json, `createdAt` varchar(255) not null, `updatedAt` varchar(255) not null) default character set utf8mb4 - You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'json, `createdAt` varchar(255) not null, `updatedAt` varchar(255) not null) defa' at line 1
2019-10-23T06:44:31.564Z [MASTER] error: Database Initialization Error: create table `assets` (`id` int unsigned not null auto_increment primary key, `filename` varchar(255) not null, `basename` varchar(255) not null, `ext` varchar(255) not null, `kind` enum('binary', 'image') not null default 'binary', `mime` varchar(255) not null default 'application/octet-stream', `fileSize` int unsigned comment 'In kilobytes', `metadata` json, `createdAt` varchar(255) not null, `updatedAt` varchar(255) not null) default character set utf8mb4 - You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'json, `createdAt` varchar(255) not null, `updatedAt` varchar(255) not null) defa' at line 1
```

**Comment by mohammad-mahdi-kargar** [comment]
@centell you need mysql version ```8.0```

**Comment by centell** [comment]
@mohammad-mahdi-kargar thanks :D

**Comment by andjelatomic** [comment]
Dears,

I have the other problem. I am  using the MSSQL server, on a different machine, and it was working fine with the previous version, but in the latest one there are some SQL syntax errors, like 

2020-04-02T11:29:38.416Z [MASTER] info: Database Connection Successful [ OK ]
migration file "2.2.17.js" failed
migration failed with error: UPDATE "pageHistory" h1 SET "versionDate" = COALESCE((SELECT prev."createdAt" FROM "pageHistory" prev WHERE prev."pageId" = h1."pageId" AND prev.id < h1.id ORDER BY prev.id DESC LIMIT 1), h1."createdAt") - Incorrect syntax near the keyword 'ORDER'.
2020-04-02T11:29:38.889Z [MASTER] error: Database Initialization Error: UPDATE "pageHistory" h1 SET "versionDate" = COALESCE((SELECT prev."createdAt" FROM "pageHistory" prev WHERE prev."pageId" = h1."pageId" AND prev.id < h1.id ORDER BY prev.id DESC LIMIT 1), h1."createdAt") - Incorrect syntax near the keyword 'ORDER'.
root@wiki01:~/wiki# 

Is there anywhere debugged version, or I have to debug it on my own?

Winds, galaxies and regards
Andjela

**Comment by hugomarshall** [comment]
> Dears,
> 
> I have the other problem. I am using the MSSQL server, on a different machine, and it was working fine with the previous version, but in the latest one there are some SQL syntax errors, like
> 
> 2020-04-02T11:29:38.416Z [MASTER] info: Database Connection Successful [ OK ]
> migration file "2.2.17.js" failed
> migration failed with error: UPDATE "pageHistory" h1 SET "versionDate" = COALESCE((SELECT prev."createdAt" FROM "pageHistory" prev WHERE prev."pageId" = h1."pageId" AND prev.id < h1.id ORDER BY prev.id DESC LIMIT 1), h1."createdAt") - Incorrect syntax near the keyword 'ORDER'.
> 2020-04-02T11:29:38.889Z [MASTER] error: Database Initialization Error: UPDATE "pageHistory" h1 SET "versionDate" = COALESCE((SELECT prev."createdAt" FROM "pageHistory" prev WHERE prev."pageId" = h1."pageId" AND prev.id < h1.id ORDER BY prev.id DESC LIMIT 1), h1."createdAt") - Incorrect syntax near the keyword 'ORDER'.
> root@wiki01:~/wiki#
> 
> Is there anywhere debugged version, or I have to debug it on my own?
> 
> Winds, galaxies and regards
> Andjela

Andjela, 
Did you solve your problem? I have a similar problem in a new installation...

**Comment by NGPixel** [comment]
@hugomarshall This has been fixed in 2.3, currently in beta.
